### PR TITLE
Rover: Add pivot_turn_rate parameter 

### PR DESCRIPTION
**Pull request by Ammarf**
This PR fixes [this issue](https://github.com/ArduPilot/ardupilot/issues/8404) found in [this discussion](https://discuss.ardupilot.org/t/skid-steer-mower-overshooting-pivot-turns/28910/18).

### CODE REVIEW

**Comment by rmackay9** [comment]
LGTM, thanks!

File: APMrover2/Parameters.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ class Parameters {
         k_param_pivot_turn_angle,
         k_param_rc_13_old,  // unused
         k_param_rc_14_old,  // unused
+        k_param_pivot_turn_rate,
```
Comment (khancyr) [inline APMrover2/Parameters.h]: I don't think this is good as k_param_pivot_turn_rate is 20 and 20 is already taken by k_param_rssi_pin 
Better put the parameter in g2

Comment (rmackay9) [reply APMrover2/Parameters.h]: nice catch @khancyr!

Comment (Ammarf) [reply APMrover2/Parameters.h]: fixed, Thanks!

**Comment by khancyr** [comment]
the first commit message is more something like : 
AR_AttitudeControl: Add limits to desired steering rate 

Squash the second and third commit and it should be good !

**Comment by rmackay9** [comment]
Merged, nice!

**Pull request merged**